### ISSUE TITLE
undefined method `connect' for Redis:Class

### ISSUE THREAD

**Issue by pengpengmao** [delano/redis-dump#19]
ERROR (NoMethodError): undefined method `connect' for Redis:Class When "redis-dump" is executed

**Comment by Crackpot** [comment]
I also met it
ERROR (NoMethodError): undefined method `connect' for Redis:Class

**Comment by ustczz** [comment]
anybody?

**Comment by MinghuiChen** [comment]
mee too:
ERROR (NoMethodError): undefined method `connect' for Redis:Class

**Comment by ilantipov** [comment]
Same for me. Anyone solved the problem?

**Comment by vakuum** [comment]
This problem is caused by the redis 4.0 gem.

Relevant part from the [CHANGELOG.md](https://github.com/redis/redis-rb/blob/master/CHANGELOG.md):

> Removed Redis.connect. Use Redis.new.

## Reproduction

```
$ rm -rf /tmp/redis-dump

$ export GEM_HOME=/tmp/redis-dump

$ export GEM_PATH=/tmp/redis-dump

$ gem install redis-dump --version 0.3.5
...
Fetching: redis-4.0.1.gem (100%)
Successfully installed redis-4.0.1
...
5 gems installed

$ $GEM_HOME/bin/redis-dump
ERROR (NoMethodError): undefined method `connect' for Redis:Class
```

## Workaround

Install redis 3.3.5 before redis-dump 0.3.5:

```
$ rm -rf /tmp/redis-dump

$ export GEM_HOME=/tmp/redis-dump

$ export GEM_PATH=/tmp/redis-dump

$ gem install redis --version 3.3.5
Fetching: redis-3.3.5.gem (100%)
Successfully installed redis-3.3.5
1 gem installed

$ gem install redis-dump --version 0.3.5
...
4 gems installed

$ $GEM_HOME/bin/redis-dump
...
```

## Versions

```
$ ruby --version
ruby 2.2.8p477 (2017-09-14 revision 59906) [x86_64-linux]

$ gem --version
2.6.14
```

**Comment by Crackpot** [comment]
Thank you very much. It solves the problem

root@crackpot-Ideapad-V460 [23:54:31] [~] 
-> # gem uninstall redis  

Select gem to uninstall:
 1. redis-3.3.5
 2. redis-4.0.1
 3. All versions
> 2
Successfully uninstalled redis-4.0.1

root@crackpot-Ideapad-V460 [23:54:56] [~] 
-> # redis-dump                     
NOAUTH Authentication required.

**Comment by vakuum** [comment]
I created a pull request for this issue: #20.

**Comment by delano** [comment]
Fix in 0.3.6. Thanks for the report and fix.
